contributions/views.py

```diff
@@ -1,158 +1,266 @@
 
-        amount  = amount_item.get('Value')
-        receipt = receipt_item.get('Value')
-        phone   = str(phone_item.get('Value'))
+        #+for update_status we want to fetch any contribution
+        if self.action == 'update_status':
+            return qs
 
-        # Validate required fields
-        if not all([amount, receipt, phone]):
-            return Response({'error': 'Missing required fields'}, status=400)
+        # allow filtering by chama for non-staff
+        chama_id = self.request.query_params.get('chama')
+        if chama_id:
+            qs = qs.filter(schedule__chama_id=chama_id)
 
-        # Prevent double-saving M-Pesa receipts
-        if Contribution.objects.filter(reference=receipt, method='MPESA').exists():
-            return Response({'message': 'Transaction already processed'}, status=200)
+        if self.request.user.is_staff:
+            return qs
+
+        # non-staff users must belong to the chama
+        if not chama_id:
+            return Contribution.objects.none()
 
-        # Get user by phone - return 404 if not found
         try:
-            user = get_user_by_phone(phone)
-        except Exception:
-            return Response({'error': 'User not found'}, status=404)
-        
-        if not user:
-            return Response({'error': 'User not found'}, status=404)
+            membership = Membership.objects.get(
+                user=self.request.user,
+                chama_id=chama_id
+            )
+        except Membership.DoesNotExist:
+            return Contribution.objects.none()
+
+        # ADMIN/TREASURER sees all; MEMBER sees only their own
+        if membership.role in (
+            Membership.Role.ADMIN,
+            Membership.Role.TREASURER
+        ):
+            return qs
+        return qs.filter(member=membership)
 
+    def create(self, request, *args, **kwargs):
+        """
+        - Manual contributions: method=CASH, status=PENDING  
+        - M-Pesa in callback only; clients cannot create MPESA here  
+        """
+        serializer = self.get_serializer(data=request.data)
+        serializer.is_valid(raise_exception=True)
+
+        schedule = serializer.validated_data['schedule']
+        amount   = serializer.validated_data['amount']
+        method   = serializer.validated_data.get('method') or Contribution.PaymentMethod.CASH
+
+        # verify membership
         try:
-            chama = get_chama_for_user(user)
-        except Exception:
-            return Response({'error': 'Chama not found for user'}, status=404)
-        
-        if not chama:
-            return Response({'error': 'Chama not found for user'}, status=404)
+            membership = Membership.objects.get(
+                user=request.user,
+                chama=schedule.chama
+            )
+        except Membership.DoesNotExist:
+            raise PermissionDenied("You are not a member of this Chama.")
```

Comment:
Use the `IsChamaMember` permission instead of manually checking for permission in the view.